src/extensions.rs

```diff
@@ -195,4 +195,27 @@ impl<T: AsRef<str>> T {
         let index = s.find(c)?;
         Some((&s[..index], &s[index + c.len_utf8()..]))
     }
+
+    fn split_at_word<'a>(&'a self, split_at: &str) -> (String, String) {
+        let str: &str = self.as_ref();
+        let mut result = (String::from(""), String::from(""));
+        let mut splits = str.trim().split(' ').collect_vec();
+        match splits.last() {
```

Comment:
Does this need special handling? not really, no? I'd think that just sth like 
```
let mut words = self.as_ref().trim().split(' ').collect_vec();
match words.iter().position(|word| w == &split_at) {
  Some(word_idx) => {
    let right_side = words.split_off(word_idx + 1).join(" ");
    (words.join(" "), right_side)
  }
  None => (self.to_string(), String::new())
}
```
should do the trick, no?